larq_compute_engine/tflite/kernels/bmaxpool.cc

```diff
@@ -122,30 +78,10 @@ TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
   TfLiteTensor* output = GetOutput(context, node, 0);
   const TfLiteTensor* input = GetInput(context, node, 0);
 
-  const TBitpacked* packed_input_data;
   RuntimeShape packed_input_shape;
 
-  // Note: When we update the bconv op with an optimized path for bitpacked
-  // input then we might have to change the bitpack order here from Canonical to
-  // Optimized
-  if (input->type == kTfLiteFloat32) {
-    TfLiteTensor* packed_input = GetTemporary(context, node, 0);
-    ce::core::bitpack_tensor(GetTensorShape(input), GetTensorData<float>(input),
-                             0, packed_input_shape,
-                             GetTensorData<TBitpacked>(packed_input));
-    packed_input_data = GetTensorData<TBitpacked>(packed_input);
-  } else if (input->type == kTfLiteInt8) {
-    TfLiteTensor* packed_input = GetTemporary(context, node, 0);
-    ce::core::bitpack_tensor(GetTensorShape(input),
-                             GetTensorData<std::int8_t>(input),
-                             input->params.zero_point, packed_input_shape,
-                             GetTensorData<TBitpacked>(packed_input));
-    packed_input_data = GetTensorData<TBitpacked>(packed_input);
-  } else {
-    TF_LITE_ENSURE_EQ(context, input->type, kTfLiteInt32);
-    packed_input_shape.ReplaceWith(4, GetTensorShape(input).DimsData());
-    packed_input_data = GetTensorData<TBitpacked>(input);
-  }
+  packed_input_shape.ReplaceWith(4, GetTensorShape(input).DimsData());
+  const TBitpacked* packed_input_data = GetTensorData<TBitpacked>(input);
 
   ce::ref::BMaxPool(*poolparams, packed_input_shape, packed_input_data,
                     GetTensorShape(output), GetTensorData<TBitpacked>(output));
```

Comment:
Now you can even remove the lines with `packed_input_shape` and `packed_input_data` (above the lines in this suggestion) entirely :smiley: 
```suggestion
  ce::ref::BMaxPool(*poolparams, GetTensorShape(input), GetTensorData<TBitpacked>(input),
                    GetTensorShape(output), GetTensorData<TBitpacked>(output));
```

Reply:
Good point, that's a lot nicer :+1: